src/Microsoft.Extensions.Configuration.AzureAppConfiguration/AzureAppConfigurationProvider.cs

```diff
@@ -33,8 +32,11 @@ internal class AzureAppConfigurationProvider : ConfigurationProvider, IConfigura
         private AzureAppConfigurationOptions _options;
         private Dictionary<string, ConfigurationSetting> _mappedData;
         private Dictionary<KeyValueIdentifier, ConfigurationSetting> _watchedSettings = new Dictionary<KeyValueIdentifier, ConfigurationSetting>();
+        private Dictionary<KeyValueIdentifier, IEnumerable<MatchConditions>> _watchedSelectedKeyValueCollections = new Dictionary<KeyValueIdentifier, IEnumerable<MatchConditions>>();
+        private Dictionary<KeyValueIdentifier, IEnumerable<MatchConditions>> _watchedFeatureFlagCollections = new Dictionary<KeyValueIdentifier, IEnumerable<MatchConditions>>();
         private RequestTracingOptions _requestTracingOptions;
         private Dictionary<Uri, ConfigurationClientBackoffStatus> _configClientBackoffs = new Dictionary<Uri, ConfigurationClientBackoffStatus>();
+        private List<KeyValueWatcher> _selectedKeyValueWatchers = new List<KeyValueWatcher>();
```

Comment:
Instead of having selectedKeyValueWatchers here, I think it will be more consistent to follow the pattern we had for feature flags.

We can add `SelectedKeyValueWatchers` in `AzureAppConfigurationOptions`. And in the `ConfigureRefresh` method, we can check whether RegisterAll is called, if yes, we create KeyValueWatcher for all kv selectors and add them to `SelectedKeyValueWatchers`.

Reply:
Removed this property in favor of a single refresh time tracked in the provider for `RegisterAll`